packages/sdk/src/oracle/services/optimisticOracle.ts

```diff
@@ -1,66 +1,108 @@
 import { optimisticOracle } from "../../clients";
-import { BigNumberish, BigNumber, Provider, Signer, TransactionResponse, Log } from "../types/ethers";
-import type { RequestState } from "../types/state";
+import { BigNumberish, Provider, Signer, TransactionResponse, Log, TransactionReceipt } from "../types/ethers";
+import type { OracleInterface, RequestKey, OracleProps, Request } from "../types/interfaces";
+import { requestId, insertOrderedAscending, eventKey, isUnique } from "../utils";
 
-type Props = {
-  defaultLiveness: BigNumber;
-};
-type Request = ReturnType<optimisticOracle.Instance["getRequest"]>;
+import { RequestPrice, ProposePrice, DisputePrice, Settle } from "../../clients/optimisticOracle";
 
-export class OptimisticOracle {
-  public readonly contract: optimisticOracle.Instance;
-  constructor(protected provider: Provider, protected address: string) {
+export type OptimisticOracleEvent = RequestPrice | ProposePrice | DisputePrice | Settle;
+
+export type { RequestPrice, ProposePrice, DisputePrice, Settle };
+
+export class OptimisticOracle implements OracleInterface {
```

Comment:
this class is the bulk of the new logic changes. it updates the oracle client to be stateful, and handle a lot of its own processing internally. It conforms to the oracle interface.